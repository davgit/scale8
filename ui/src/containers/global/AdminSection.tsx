import { FC } from 'react';
import { useQuery } from '@apollo/client';
import SignUpApprovalIcon from '../../components/atoms/Icons/SignUpApprovalIcon';
import AdminDashboardIcon from '../../components/atoms/Icons/AdminDashboardIcon';
import NavAdminQuery from '../../gql/queries/NavAdminQuery';
import { NavAdmin } from '../../gql/generated/NavAdmin';
import { buildAdminButtonProps, buildTabButtonProps } from '../../utils/BreadcrumbButtonsUtils';
import { Section, SectionProps } from '../abstractions/Section';
import { SectionKey } from '../SectionsDetails';
import { ChildrenOnlyProps } from '../../types/props/ChildrenOnlyProps';
import { toAdmin, toSignupApproval } from '../../utils/NavigationPaths';
import { useRouter } from 'next/router';
import { PageMenuButtonProps } from '../../components/molecules/SideMenuButton';

export const buildAdminTabsMenu = (): PageMenuButtonProps[] => [
    {
        icon: () => <AdminDashboardIcon />,
        label: 'Dashboard',
        link: toAdmin,
    },
    {
        icon: () => <SignUpApprovalIcon />,
        label: 'Signup Approval',
        link: toSignupApproval,
    },
];

const AdminSection: FC<ChildrenOnlyProps> = (props: ChildrenOnlyProps) => {
    const router = useRouter();

    const sectionProps: SectionProps<NavAdmin> = {
        // Admin = all permissions
        extractOrgUserDetails: () => ({
            id: '',
            name: '',
            is_paid: false,
            tag_manager_account: {
                id: '',
                trial_expires_in: 0,
                trial_expired: false,
                is_trial: false,
                enabled: true,
            },
            data_manager_account: {
                id: '',
                trial_expires_in: 0,
                trial_expired: false,
                is_trial: false,
                enabled: true,
            },
            me: {
                id: '',
                owner: false,
                permissions: {
                    can_view: true,
                    can_create: true,
                    can_edit: true,
                    can_delete: true,
                    is_admin: true,
                },
            },
        }),
        sectionKey: SectionKey.admin,
        queryResult: useQuery<NavAdmin>(NavAdminQuery),
        buildButtonsProps: (data) => [
            buildAdminButtonProps(router, data.me.orgs),
            buildTabButtonProps(router, buildAdminTabsMenu()),
        ],
        buildMenuItemsProps: () => buildAdminTabsMenu(),
        children: props.children,
    };

    return <Section<NavAdmin> {...sectionProps} />;
};

export default AdminSection;
