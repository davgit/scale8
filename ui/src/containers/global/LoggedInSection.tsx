import * as React from 'react';
import { FC, MouseEventHandler, useEffect, useReducer, useState } from 'react';
import LoggedInTemplate from '../../components/templates/containers/LoggedInTemplate';
import { ApolloError, useQuery } from '@apollo/client';
import { LoggedUser } from '../../gql/generated/LoggedUser';
import LoggedUserQuery from '../../gql/queries/LoggedUserQuery';
import { createHash } from 'crypto';
import { useCancelConfirmDialog } from '../../hooks/dialog/useCancelConfirmDialog';
import { CancelConfirmDialogProps } from '../../types/props/CancelConfirmDialogProps';
import { pageActions } from '../../actions/PageActions';
import { dialogReducer } from '../../context/DialogReducer';
import { orgUserReducer } from '../../context/OrgUserReducer';
import GithubCheck from '../../dialogPages/global/GithubCheck';
import { ChildrenOnlyProps } from '../../types/props/ChildrenOnlyProps';
import Navigate from '../../components/atoms/Next/Navigate';
import { dialogInit } from '../../context/LoggedInState';
import { useAppContext } from '../../context/AppContext';
import Loader from '../../components/organisms/Loader';
import { configStateFromData } from '../../context/ConfigState';
import { SectionKey } from '../SectionsDetails';
import { clearAuthSession } from '../../utils/authUtils';

export type UserSelectorProps = {
    loading: boolean;
    imgSrc?: string;
    userName: string;
    userEmail: string;
    handleAccountClick: MouseEventHandler;
    handleLogoutClick: MouseEventHandler;
};

export type SideBarProps = {
    handleNotificationClick: MouseEventHandler;
    children?: React.ReactNode;
    userSelectorProps: UserSelectorProps;
    isAdmin: boolean;
    unreadNotifications: number;
};

export type LoggedInProps = {
    sideBarProps: SideBarProps;
    children: React.ReactNode;
    breadcrumb: React.ReactNode;
    sideMenu: React.ReactNode;
    cancelConfirmDialogProps: CancelConfirmDialogProps;
};

const LoggedInSection: FC<ChildrenOnlyProps> = (props: ChildrenOnlyProps) => {
    const { setState, loggedInState, configState } = useAppContext();

    const [logOut, setLogOut] = useState(false);

    const [dialogState, dispatchDialogAction] = useReducer(dialogReducer, null, dialogInit);

    const [refreshCurrentPage, setRefreshCurrentPage] = useState(false);
    const [refreshCurrentSection, setRefreshCurrentSection] = useState(false);

    const [snackbarError, setSnackbarError] = useState<ApolloError | null>(null);

    const [orgUserState, dispatchOrgUserAction] = useReducer(orgUserReducer, null);

    const [section, setSection] = useState<symbol>(SectionKey.loggedOut);

    const [sectionHasAnalytics, setSectionHasAnalytics] = useState(false);

    const [gates, dispatchGatesAction] = useReducer(
        (
            state: Record<string, JSX.Element>,
            action: { type: 'set'; payload: { gateName: string; element: JSX.Element } },
        ): Record<string, JSX.Element> => {
            return { ...state, [action.payload.gateName]: action.payload.element };
        },
        {},
    );

    const teleport = (gateName: string, element: JSX.Element) => {
        dispatchGatesAction({ type: 'set', payload: { gateName, element } });
    };

    const { loading, error, data, ...queryResult } = useQuery<LoggedUser>(LoggedUserQuery);

    // Single Question Dialog
    const { dialogProps: cancelConfirmDialogProps, ask } = useCancelConfirmDialog();

    useEffect(() => {
        setState({
            loggedInState: {
                templateInteractions: {
                    ask,
                    dialogState,
                    dispatchDialogAction,
                    snackbarError,
                    setSnackbarError,
                    refreshCurrentPage,
                    setRefreshCurrentPage,
                    refreshCurrentSection,
                    setRefreshCurrentSection,
                    section,
                    setSection,
                    sectionHasAnalytics,
                    setSectionHasAnalytics,
                },
                loggedInUserState: {
                    loggedUserId: data?.me.id ?? '',
                    userIsAdmin: !!data?.me.is_admin,
                    invites: data ? data.me.invites.length : 0,
                    notifications: data ? data.me.user_notifications.length : 0,
                },
                orgUserState,
                dispatchOrgUserAction,
                gates,
                teleport,
            },
            configState: configStateFromData(data),
        });
    }, [
        data,
        dialogState,
        refreshCurrentPage,
        refreshCurrentSection,
        snackbarError,
        orgUserState,
        section,
    ]);

    if (error || logOut) {
        return <Navigate to="/login" />;
    }

    if (loading || loggedInState === undefined || configState === undefined) {
        return <Loader />;
    }

    const gravatarHash = createHash('md5')
        .update(data ? data.me.email.trim().toLowerCase() : '')
        .digest('hex');

    const userSelectorProps: UserSelectorProps = {
        imgSrc: `https://www.gravatar.com/avatar/${gravatarHash}?s=100&d=identicon`,
        loading,
        userName: `${data?.me.first_name} ${data?.me.last_name}`,
        userEmail: `${data?.me.email}`,
        handleAccountClick: () => {
            pageActions.manageAccount({
                dispatchDialogAction,
                refresh: () => {
                    (async () => {
                        await queryResult.refetch();
                    })();
                },
            });
        },
        handleLogoutClick: () => {
            clearAuthSession();
            setLogOut(true);
        },
    };

    const sideBarProps: SideBarProps = {
        handleNotificationClick: () => {
            pageActions.showNotification({
                dispatchDialogAction,
            });
        },
        userSelectorProps,
        isAdmin: data ? data.me.is_admin : false,
        unreadNotifications: data ? data.me.invites.length + data.me.user_notifications.length : 0,
    };

    return (
        <>
            <GithubCheck
                githubUser={data?.me.github_user ?? null}
                githubConnected={data?.me.github_connected ?? false}
            />
            <LoggedInTemplate
                sideBarProps={sideBarProps}
                breadcrumb={gates['breadcrumb'] ?? <></>}
                sideMenu={gates['side'] ?? <></>}
                cancelConfirmDialogProps={cancelConfirmDialogProps}
            >
                {props.children}
            </LoggedInTemplate>
        </>
    );
};

export default LoggedInSection;
