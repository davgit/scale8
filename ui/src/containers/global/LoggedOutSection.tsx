import { FC, useEffect } from 'react';
import LoggedOutTemplate from '../../components/templates/containers/LoggedOutTemplate';
import { ChildrenOnlyProps } from '../../types/props/ChildrenOnlyProps';
import { clearAuthSession } from '../../utils/authUtils';

const LoggedOutSection: FC<ChildrenOnlyProps & { stayLogged?: boolean }> = ({
    children,
    stayLogged,
}) => {
    useEffect(() => {
        if (!stayLogged) {
            clearAuthSession();
        }
    }, [stayLogged]);

    return <LoggedOutTemplate>{children}</LoggedOutTemplate>;
};

export default LoggedOutSection;
