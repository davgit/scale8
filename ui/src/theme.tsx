import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
