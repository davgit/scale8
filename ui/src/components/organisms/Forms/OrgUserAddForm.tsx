import { FC } from 'react';
import ControlledTextInput from '../../atoms/ControlledInputs/ControlledTextInput';
import FormPermissionSection from '../../molecules/FormPermissionSection';
import DrawerFormLayout from '../../molecules/DrawerFormLayout';
import { OrgUserAddFormProps } from '../../../dialogPages/global/OrgUserAdd';

const OrgUserAddForm: FC<OrgUserAddFormProps> = (props: OrgUserAddFormProps) => {
    return (
        <DrawerFormLayout {...props}>
            <ControlledTextInput
                name="firstName"
                label="First Name"
                formProps={props}
                className="DialogFormField"
                required
                fullWidth
            />
            <ControlledTextInput
                name="lastName"
                label="Last Name"
                formProps={props}
                className="DialogFormField"
                required
                fullWidth
            />
            <ControlledTextInput
                name="email"
                label="Email"
                formProps={props}
                className="DialogFormField"
                required
            />
            <FormPermissionSection formProps={props} />
        </DrawerFormLayout>
    );
};

export default OrgUserAddForm;
