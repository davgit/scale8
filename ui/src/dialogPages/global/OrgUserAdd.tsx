import { FC } from 'react';
import { ApolloError, useMutation } from '@apollo/client';
import {
    permissionsCustomValueSetter,
    permissionsInitialState,
    PermissionsValues,
} from '../../utils/PermissionsUtils';
import { FormProps, FormValidationResult } from '../../hooks/form/useFormValidation';
import emailValidator from '../../utils/validators/emailValidator';
import { buildStandardFormInfo } from '../../utils/InfoLabelsUtils';
import { DialogPageProps } from '../../types/DialogTypes';
import OrgAddUserQuery from '../../gql/mutations/OrgAddUserQuery';
import { AddOrgUserResult } from '../../gql/generated/AddOrgUserResult';
import OrgUserAddForm from '../../components/organisms/Forms/OrgUserAddForm';
import { DialogForm, DialogFormProps } from '../abstractions/DialogForm';

export type OrgUserAddValues = PermissionsValues & {
    firstName: string;
    lastName: string;
    email: string;
};

export type OrgUserAddFormProps = FormProps<OrgUserAddValues>;

const OrgUserAdd: FC<DialogPageProps> = (props: DialogPageProps) => {
    const orgId = props.contextId;

    const orgUserAddProps: DialogFormProps<
        OrgUserAddValues,
        OrgUserAddFormProps,
        AddOrgUserResult
    > = {
        buildInitialState: () => ({
            firstName: '',
            lastName: '',
            email: '',
            ...permissionsInitialState,
        }),
        saveQuery: useMutation<AddOrgUserResult>(OrgAddUserQuery),
        mapSaveData: (orgUserAddValues: OrgUserAddValues) => ({
            orgAddUserInput: {
                org_id: orgId,
                email: orgUserAddValues.email,
                first_name: orgUserAddValues.firstName,
                last_name: orgUserAddValues.lastName,
                org_permission_group: {
                    can_view: orgUserAddValues.permissionsCanView,
                    can_create: orgUserAddValues.permissionsCanCreate,
                    can_edit: orgUserAddValues.permissionsCanEdit,
                    can_delete: orgUserAddValues.permissionsCanDelete,
                    is_admin: orgUserAddValues.permissionsIsAdmin,
                },
            },
        }),
        buildFormProps: (
            formValidationValues: FormValidationResult<OrgUserAddValues>,
            gqlError?: ApolloError,
        ) => ({
            ...formValidationValues,
            gqlError,
            submitText: 'Add User',
            title: 'Add User',
            formInfoProps: buildStandardFormInfo('organizationUsers', 'Add'),
            handleDialogClose: props.handleDialogClose,
        }),
        checkSuccessfullySubmitted: (formMutationData) => !!formMutationData?.addUser.id,
        getSavedId: (formMutationData) => formMutationData?.addUser.id,
        pageComponent: OrgUserAddForm,
        validators: [
            {
                field: 'email',
                validator: emailValidator,
                error: () => 'Invalid email address',
            },
        ],
        customValueSetter: permissionsCustomValueSetter,
        ...props,
    };

    return (
        <DialogForm<OrgUserAddValues, OrgUserAddFormProps, AddOrgUserResult> {...orgUserAddProps} />
    );
};

export default OrgUserAdd;
