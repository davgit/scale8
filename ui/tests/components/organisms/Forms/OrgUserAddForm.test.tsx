import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import OrgUserAddForm from '../../../../src/components/organisms/Forms/OrgUserAddForm';
import { OrgUserAddValues } from '../../../../src/dialogPages/global/OrgUserAdd';

const MockForm: FC = () => {
    const formProps = useMockFormProps<OrgUserAddValues>({
        firstName: '',
        lastName: '',
        email: '',
        permissionsCanCreate: false,
        permissionsCanDelete: false,
        permissionsCanEdit: false,
        permissionsCanView: true,
        permissionsIsAdmin: false,
    });

    return <OrgUserAddForm {...formProps} />;
};

describe('OrgUserAddForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
