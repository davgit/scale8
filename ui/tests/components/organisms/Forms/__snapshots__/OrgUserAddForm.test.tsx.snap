// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrgUserAddForm form render unchanged 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-s92abg"
  >
    <div
      class="MuiBox-root css-70qvj9"
    >
      <div
        class="MuiBox-root css-1isemmb"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-uix0gv"
      >
        mock form
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-yn9cjk"
    >
      <div
        class="MuiBox-root css-1676va5"
      >
        <form
          class="MuiBox-root css-mjj15b"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root DialogFormField css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="mui-1"
              id="mui-1-label"
            >
              First Name
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-styb2u-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id="mui-1"
                name="firstName"
                required=""
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root DialogFormField css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="mui-2"
              id="mui-2-label"
            >
              Last Name
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-styb2u-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id="mui-2"
                name="lastName"
                required=""
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="mui-3"
              id="mui-3-label"
            >
              Email
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-b4y4zd-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1v2pam7-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="off"
                class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id="mui-3"
                name="email"
                required=""
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                    data-testid="ClearIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-l763sf"
          >
            <div
              class="MuiBox-root css-tet811"
            >
              User Permissions
            </div>
            <fieldset
              class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
            >
              <div
                class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
              >
                <label
                  class="MuiFormControlLabel-root Mui-disabled MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
                  >
                    <span
                      aria-disabled="true"
                      class="MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-disabled MuiButtonBase-root Mui-disabled MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-disabled PrivateSwitchBase-root Mui-checked Mui-disabled css-z5i0yo-MuiButtonBase-root-MuiSwitch-switchBase"
                      tabindex="-1"
                    >
                      <input
                        checked=""
                        class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                        disabled=""
                        name="permissionsCanView"
                        type="checkbox"
                      />
                      <span
                        class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                      />
                    </span>
                    <span
                      class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label Mui-disabled css-13p0dqt-MuiTypography-root"
                  >
                    Can View
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
                  >
                    <span
                      class="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-z5i0yo-MuiButtonBase-root-MuiSwitch-switchBase"
                    >
                      <input
                        class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                        name="permissionsCanCreate"
                        type="checkbox"
                      />
                      <span
                        class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                      />
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
                  >
                    Can Create
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
                  >
                    <span
                      class="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-z5i0yo-MuiButtonBase-root-MuiSwitch-switchBase"
                    >
                      <input
                        class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                        name="permissionsCanEdit"
                        type="checkbox"
                      />
                      <span
                        class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                      />
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
                  >
                    Can Edit
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
                  >
                    <span
                      class="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-z5i0yo-MuiButtonBase-root-MuiSwitch-switchBase"
                    >
                      <input
                        class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                        name="permissionsCanDelete"
                        type="checkbox"
                      />
                      <span
                        class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                      />
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
                  >
                    Can Delete
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
                  >
                    <span
                      class="MuiSwitch-switchBase MuiSwitch-colorPrimary MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root css-z5i0yo-MuiButtonBase-root-MuiSwitch-switchBase"
                    >
                      <input
                        class="MuiSwitch-input PrivateSwitchBase-input css-1m9pwf3"
                        name="permissionsIsAdmin"
                        type="checkbox"
                      />
                      <span
                        class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                      />
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
                  >
                    Is Admin
                  </span>
                </label>
              </div>
            </fieldset>
          </div>
          <div
            class="MuiBox-root css-wpygli"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-xcbv7o-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
